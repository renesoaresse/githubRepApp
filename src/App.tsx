import React from 'react';
import {SafeAreaView, StatusBar} from 'react-native';
import Routers from './routers';
import {Provider} from 'react-redux';
import {store} from './store/store';

const App = () => (
  <SafeAreaView style={{flex: 1, backgroundColor: 'white'}}>
    <StatusBar barStyle={'light-content'} />
    <Provider store={store}>
      <Routers />
    </Provider>
  </SafeAreaView>
);

export default App;
