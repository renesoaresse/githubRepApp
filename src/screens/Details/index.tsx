import React from 'react';
import {WebView} from 'react-native-webview';
import {RouteProp} from '@react-navigation/native';

type Props = {
  route: RouteProp<{params: {html_url: string}}, 'params'>;
};

export const Details = ({route}: Props) => {
  const {html_url} = route.params;

  return <WebView source={{uri: html_url}} />;
};
