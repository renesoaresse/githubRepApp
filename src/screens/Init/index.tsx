import React, {useState} from 'react';
import {
  TextInput,
  Text,
  Image,
  View,
  FlatList,
  TouchableOpacity,
} from 'react-native';

import {StackNavigationProp} from '@react-navigation/stack';
import {useSelector, useDispatch} from 'react-redux';

import api from '../../services/api';
import {RootState} from '../../store/store';
import {setRepositories} from '../../store/slices/repository';

type Props = {
  navigation: StackNavigationProp<any>;
};

export const Init = ({navigation}: Props) => {
  const [text, setText] = useState('');
  const dispatch = useDispatch();
  const repositories = useSelector(
    (state: RootState) => state.repository.repositories,
  );
  console.log('ðŸš€ ~ file: index.tsx:28 ~ Init ~ repositories:', repositories);

  const searchRepos = async () => {
    const {data} = await api.get(`/search/repositories?q=${text}&per_page=1`);
    dispatch(setRepositories(data.items));
  };

  return (
    <View style={{alignContent: 'center', justifyContent: 'center'}}>
      <TextInput
        value={text}
        onChangeText={setText}
        onBlur={searchRepos}
        onSubmitEditing={searchRepos}
        style={{borderWidth: 1}}
      />
      <FlatList
        data={repositories}
        renderItem={({item}) => (
          <View>
            <TouchableOpacity
              onPress={() =>
                navigation.navigate('DetailsScreen', {
                  html_url: item.owner.html_url,
                })
              }>
              <Image
                source={{uri: item.owner.avatar_url}}
                style={{width: 52, height: 52, borderRadius: 25}}
              />
              <Text>{item.name}</Text>
              <Text>{item.owner.login}</Text>
              <Text>{item.stargazers_count}</Text>
            </TouchableOpacity>
          </View>
        )}
        keyExtractor={item => String(item.id)}
      />
    </View>
  );
};
