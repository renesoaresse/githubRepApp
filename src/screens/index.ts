export {Init} from './Init';
export {Details} from './Details';
