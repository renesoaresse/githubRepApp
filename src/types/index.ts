export type {
  RepositoryInterface,
  RepositoryStateInterface,
} from './RepositoryInterface';
