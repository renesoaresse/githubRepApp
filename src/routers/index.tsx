import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {Init, Details} from '../screens';

type StackParamList = {
  Init: undefined;
  Details: {title: string};
};

const Stack = createNativeStackNavigator<StackParamList>();

const Routers = () => (
  <NavigationContainer>
    <Stack.Navigator
      screenOptions={{
        headerStyle: {
          backgroundColor: '#fff',
        },
        headerTintColor: '#000',
        headerTitleStyle: {
          fontWeight: 'bold',
        },
      }}>
      <Stack.Screen name="Init" component={Init} />
      <Stack.Screen
        name="Details"
        component={Details}
        options={({route}) => ({title: route.params.title})}
      />
    </Stack.Navigator>
  </NavigationContainer>
);
export default Routers;
