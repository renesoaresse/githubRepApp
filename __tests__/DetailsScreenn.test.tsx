import React from 'react';
import {render} from '@testing-library/react-native';

import {Details} from '../src/screens';

jest.mock('react-native-webview', () => {
  const {View} = require('react-native');
  return {
    WebView: (props: any) => {
      return <View testID="mockWebView" {...props} />;
    },
  };
});

describe('Details', () => {
  it('renders WebView with correct source', () => {
    const html_url = 'https://github.com';
    const {getByTestId, toJSON} = render(
      <Details
        route={{
          key: 'test',
          name: 'Details',
          params: {html_url},
        }}
      />,
    );

    const webView = getByTestId('mockWebView');
    expect(webView.props.source).toEqual({uri: html_url});

    expect(toJSON()).toMatchSnapshot();
  });
});
