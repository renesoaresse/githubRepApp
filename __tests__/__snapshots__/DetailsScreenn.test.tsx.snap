// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders WebView with correct source 1`] = `
<View
  source={
    {
      "uri": "https://github.com",
    }
  }
  testID="mockWebView"
/>
`;
